overlord/snapstate/handlers.go

```diff
@@ -1630,6 +1630,8 @@ func (m *SnapManager) maybeRestart(t *state.Task, info *snap.Info, rebootRequire
 
 	typ := info.Type()
 
+	// TODO: even knowing that "bp" stands for "boot participant" doesn't help
+	// in making this comment more clear:
 	// if bp is non-trivial then either we're not on classic, or the snap is
```

Comment:
haha yeah sorry this comment is quite un-helpful, I think this is what the comment is getting at:

```
// if the type of the snap requesting this start is non-trivial that either means we
// are on ubuntu core and the type is a base/kernel/gadget which requires a reboot of the 
// system, or that the type is snapd in which case we just do a restart of snapd itself.
// In these cases restartReason will be non-empty and thus we will perform a restart.
// If restartReason is empty, then the snap requesting the restart was not a boot participant
// and thus we don't need to do any sort of restarts as a result of updating this snap.
```

Additionally, the doc-comment for `maybeRestart` should indicate that this function is called for all snap refreshes, not just those that may trigger a restart, and if rebootRequired is true then we know a-priori when calling this function that a reboot is needed and thus this function just always does the reboot, otherwise this function will figure out if a reboot is needed depending on the type of the snap that was just updated/linked.

This doesn't need to be updated in this PR to be clear, we can fix this in a different one, your choice

Reply:
Applied, thanks!